compiler/rustc_session/src/session.rs

```diff
@@ -1695,6 +1695,25 @@ fn validate_commandline_args_with_session_available(sess: &Session) {
             sess.emit_err(errors::IncompatibleLinkerFlavor { flavor, compatible_list });
         }
     }
+
+    if sess.opts.unstable_opts.function_return != FunctionReturn::default() {
+        if sess.target.arch != "x86" && sess.target.arch != "x86_64" {
+            sess.emit_err(errors::FunctionReturnRequiresX86OrX8664);
+        }
+    }
+
+    // This applies to `thunk` and `thunk-extern`, but not `thunk-inline`, so it is kept as a
+    // `match` to force a change even if we only support `thunk-extern` like Clang.
+    match sess.opts.unstable_opts.function_return {
+        FunctionReturn::Keep => (),
+        FunctionReturn::ThunkExtern => {
+            // FIXME: In principle, the inherited base LLVM target code model could be large,
+            // but this only checks whether we were passed one explicitly (like Clang does).
```

Comment:
If there is a way to get the effective code model, it would be nice to use it.

Reply:
Maybe check [`sess.target.options.code_model`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_target/spec/struct.TargetOptions.html#structfield.code_model) ?

Reply:
That covers even less, no? i.e. `code_model()` covers codegen and target options.